governance/contract.go

```diff
@@ -80,41 +77,50 @@ func (e *ContractEngine) UpdateParams() error {
 
 // contractGetAllParamsAt sets evmCtx.BlockNumber as num
 func (e *ContractEngine) contractGetAllParamsAt(num uint64) (*params.GovParamSet, error) {
-	if e.chain == nil {
-		logger.Info("ContractEngine disabled: chain = nil")
-		return params.NewGovParamSet(), nil
+	chain := e.headerGov.BlockChain()
+	if chain == nil {
+		logger.Crit("headerGov.BlockChain() is nil")
+		return nil, errContractEngineNotReady
 	}
 
-	if !e.chainConfig.IsKoreForkEnabled(e.chain.CurrentHeader().Number) {
+	config := chain.Config()
+	if !config.IsKoreForkEnabled(new(big.Int).SetUint64(num)) {
 		logger.Info("ContractEngine disabled: hardfork block not passed")
```

Comment:
This info log is called when `CreateSnapshot` is called.. right? If so, what about moving the log level to `Trace`?

Reply:
fixed in 03d4270b along with similar nearby lines